### PR TITLE
[new release] odate (0.7)

### PR DESCRIPTION
**Pull request by hhugo**
Date & Duration Library

- Project page: <a href="https://github.com/hhugo/odate">https://github.com/hhugo/odate</a>

## Changes
* Compatibility with recent menhir

### CODE REVIEW

File: packages/odate/odate.0.7/opam [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+"""
+
+build: [
+  ["dune" "subst"] {pinned}
+  ["dune" "build" "-p" name "-j" jobs]
+]
+
+depends: [
+  "ocaml" {>= "4.07.0"}
+  "dune" {build}
```
Comment (haochenx) [inline packages/odate/odate.0.7/opam:25]: seems like it's missing lower bounds?
```suggestion
  "dune" {>= "2.2" & build}
```

File: packages/odate/odate.0.7/opam [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  "menhir" {build}
```
Comment (haochenx) [inline packages/odate/odate.0.7/opam:26]: menhir also seems to need lower bounds

```
#=== ERROR while compiling odate.0.7 ==========================================#
# context              2.2.0~alpha3~dev | linux/x86_64 | ocaml-base-compiler.4.12.1 | pinned(https://github.com/hhugo/odate/releases/download/0.7/odate-0.7.tbz)
# path                 ~/.opam/4.12/.opam-switch/build/odate.0.7
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p odate -j 71
# exit-code            1
# env-file             ~/.opam/log/odate-7-0080ad.env
# output-file          ~/.opam/log/odate-7-0080ad.out
### output ###
#       menhir src/duration_parser__mock.ml.mock (exit 2)
# (cd _build/default && /home/opam/.opam/4.12/bin/menhir src/duration_parser.mly --base src/duration_parser --infer-write-query src/duration_parser__mock.ml.mock)
# /home/opam/.opam/4.12/bin/menhir: unknown option '--infer-write-query'.
# Usage: /home/opam/.opam/4.12/bin/menhir <options> <filenames>
#   --base <basename>           Specifies a base name for the output file(s)
#   --canonical                 Construct a canonical Knuth LR(1) automaton
```

```
The following actions will be performed:
=== remove 2 packages
  - remove    menhirLib  20230608             [conflicts with dune]
  - remove    menhirSdk  20230608             [conflicts with dune]
=== downgrade 2 packages
  - downgrade dune       3.10.0 to 2.8.0      [required by odate]
  - downgrade menhir     20230608 to 20160303 [required by odate]
=== recompile 1 package
  - recompile odate      0.7 (pinned)
=== install 3 packages
  - install   conf-m4    1                    [required by ocamlfind]
  - install   ocamlbuild 0.14.0               [required by menhir]
  - install   ocamlfind  1.8.1                [required by menhir]
```

File: packages/odate/odate.0.7/opam (hunk 1 above)
Comment (raphael-proust) [reply packages/odate/odate.0.7/opam:25]: The `dune` dependencies should not be marked as `build`. I can't find the link to the issue that originally describes the reason why, but it is mentioned in the wiki: https://github.com/ocaml/opam-repository/wiki/PR-checks#dune-checks

Comment (haochenx) [reply packages/odate/odate.0.7/opam]: @hhugo I believe that `build` should not be there but a lower bound is necessary for Dune

**Comment by haochenx** [comment]
Thanks! Let's see how the CI does.

File: packages/odate/odate.0.7/opam [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+description: """
+Simple date and duration manipulation. Also implement duration printer
+based on string format. Already implemented in opalang
+[http://opalang.org/]. For documentation about the format, see :
+[http://doc.opalang.org/value/stdlib.core.date/Duration/try_generate_printer].
+"""
+
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (haochenx) [inline packages/odate/odate.0.7/opam:19]:
```suggestion
  ["dune" "subst"] {dev}
```

Comment (hhugo) [reply packages/odate/odate.0.7/opam]: I'll have dune generate the opam file in the future

**Comment by haochenx** [comment]
Thanks! This PR looks all good to me. Let's see how the CI goes.

**Comment by haochenx** [comment]
The revdep failure (terminus-hlc.0.1.0, reason: `(failed: Sorry, resolution of the request timed out.)`) should be unrelated.

**Comment by avsm** [comment]
Thanks! You may want to announce this on <https://discuss.ocaml.org>, where we have a Community category and an `announce` tag for this purpose.

File: packages/odate/odate.0.7/opam (hunk 3 above)
Comment (hhugo) [reply packages/odate/odate.0.7/opam]: @haochenx, Are you sure about this change ? It seems like dune is generating `{pinned}` and not `{dev}`

Comment (hhugo) [reply packages/odate/odate.0.7/opam]: ok

**Pull request merged**